src/Centreon/Application/Controller/Monitoring/CommentController.php

```diff
@@ -0,0 +1,346 @@
+             * At this point we made sure that the mapping will work since we validate
+             * the JSON sent with the JSON validator.
+             */
+            $host = $this->monitoringService->findOneHost($hostId);
+            if (is_null($host)) {
+                throw new EntityNotFoundException(
+                    sprintf(
+                        _('Host %d not found'),
+                        $hostId
+                    )
+                );
+            }
+            $date = ($results['date'] !== null) ? new \DateTime($results['date']) : new \DateTime('now');
+            $result = (new Comment($hostId, $results['comment']))
+                ->setDate($date);
+
+            try {
+                $this->commentService
+                    ->filterByContact($contact)
+                    ->addHostComment($result);
+            } catch (EntityNotFoundException $e) {
+                throw $e;
+            }
+        }
+
+        return $this->view(null, Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Entry point to add a comment on a service
+     *
+     * @param Request $request
+     * @param int $hostId ID of service parent (host)
+     * @param int $serviceId ID of the service
+     * @return View
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     */
+    public function addServiceComment(
+        Request $request,
+        int $hostId,
+        int $serviceId
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+        if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_SERVICE_ADD_COMMENT)) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        $results = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/Comment.json'
+        );
+
+        if (!empty($results)) {
+            /**
+             * At this point we made sure that the mapping will work since we validate
+             * the JSON sent with the JSON validator.
+             */
+            $host = $this->monitoringService->findOneHost($hostId);
+            if (is_null($host)) {
+                throw new EntityNotFoundException(
+                    sprintf(
+                        _('Host %d not found'),
+                        $hostId
+                    )
+                );
+            }
+
+            $service = $this->monitoringService->findOneService($hostId, $serviceId);
+            if (is_null($service)) {
+                throw new EntityNotFoundException(
+                    sprintf(
+                        _('Service %d not found'),
+                        $serviceId
+                    )
+                );
+            }
+            $service->setHost($host);
+
+            $date = ($results['date'] !== null) ? new \DateTime($results['date']) : new \DateTime('now');
```

Comment:
```suggestion
            $date = ($results['date'] !== null) ? new \DateTime($results['date']) : new \DateTime();
```
'now' is the default value for a new \DateTime()